Gems/Multiplayer/Code/Source/Components/NetworkDebugPlayerIdComponent.cpp

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Multiplayer/Components/NetworkDebugPlayerIdComponent.h>
+
+#include <Atom/RPI.Public/ViewportContextBus.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzFramework/Viewport/ViewportScreen.h>
+#include <AzNetworking/Framework/INetworking.h>
+
+namespace Multiplayer
+{
+    AZ_CVAR(bool, cl_debugPlayerIdRender, true, nullptr, AZ::ConsoleFunctorFlags::DontReplicate, "Whether NetworkDebugPlayerIdComponent should render the player id and connection count on-screen.");
+
+    void NetworkDebugPlayerIdComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (const auto serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<NetworkDebugPlayerIdComponent, NetworkDebugPlayerIdComponentBase>()
+                ->Version(1)
+                ->Field("translationOffset", &NetworkDebugPlayerIdComponent::m_translationOffset)
+                ->Field("scale", &NetworkDebugPlayerIdComponent::m_fontScale)
+                ->Field("color", &NetworkDebugPlayerIdComponent::m_fontColor)
+            ;
+
+            if (AZ::EditContext* editContext = serializeContext->GetEditContext())
+            {
+                editContext->Class<NetworkDebugPlayerIdComponent>("Network Debug Connection Counter", "Renders the player id as debug text over network players.")
```

Comment:
Long names don't fit super well in the entity outliner, recommend something shorter
```suggestion
                editContext->Class<NetworkDebugPlayerIdComponent>("Network Debug Player ID", "Renders the player id as debug text over network players.")
```